### PR TITLE
Send HH an email when a volunteer response is created

### PR DESCRIPTION
**Pull request by jamestran201**
Trello card: https://trello.com/c/EP1E74dz/226-notify-hh-when-volunteer-makes-direct-offer-to-provider

Send an email to hello@hospitalhero.care when Volunteer submits an Offer: 
- The "From" should be the Volunteer's name/email. 
- The subject is"Offer made to **Provider name** from **Volunteer name** (**RequestId: request number**)".
- The body of the email is the same as email to provider

### CODE REVIEW

File: app/services/provider_notifications.rb [hunk 1]
```diff
@@ -4,5 +4,10 @@ def send_provider_created_notifications(linked_token)
       ProviderMailer.with(linked_token: linked_token).request_created_email.deliver_later
       InternalMailer.with(linked_token: linked_token).request_created_email.deliver_later
     end
+
+    def send_volunteer_response_created_notifications(provider, volunteer, response)
```
Comment (tuntisz) [inline app/services/provider_notifications.rb:8]: This is great! ~Minor request: can we rename this class to `Notifications` or create a `VolunteerNotifications` class?~

Comment (tuntisz) [reply app/services/provider_notifications.rb:8]: On second thought, this makes sense as is :) Ignore the last comment

File: app/mailers/internal_mailer.rb [hunk 2]
```diff
@@ -6,4 +6,12 @@ def request_created_email
     @edit_url = HOST_AND_SCHEME + '/providers/' + params[:linked_token].token + '/edit'
     mail(template_path: 'provider_mailer', to: INTERNAL_EMAIL, from: @provider.email, reply_to: @provider.email, subject: 'Your help request has been created.')
   end
+
+  def volunteer_response_email
+    @provider = params[:provider]
+    @volunteer = params[:volunteer]
+    @response = params[:response]
+    mail(template_path: "provider_mailer", to: INTERNAL_EMAIL, from: @volunteer.email, reply_to: @volunteer.email,
```
Comment (tuntisz) [inline app/mailers/internal_mailer.rb]: I just learned that we can only send emails from known senders that we setup in the Postmark UI. So we'll have to drop the from here. More details here: https://postmarkapp.com/developer/user-guide/managing-your-account/managing-sender-signatures

File: spec/services/provider_notifications.rb [hunk 3]
```diff
@@ -11,4 +11,20 @@
...
+      provider = Provider.new(id: 1)
+      volunteer = Volunteer.new(id: 1)
+      response = Response.new(id: 1)
+
+      expect {
+        ProviderNotifications.send_volunteer_response_created_notifications(provider, volunteer, response)
+      }.to have_enqueued_job.on_queue("mailers").with(
+          "ProviderMailer", "volunteer_response_email", "deliver_now", any_args
+      ).and have_enqueued_job.on_queue('mailers').with(
+        'InternalMailer', 'volunteer_response_email', 'deliver_now', any_args
```
Comment (jamestran201) [inline spec/services/provider_notifications.rb:26]: Just curious, does Rspec or Rails convert `deliver_later` to `deliver_now` in unit tests by default?

**Comment by jamestran201** [comment]
@tuntisz Could you please review again when you have time?

File: spec/services/provider_notifications.rb (hunk 3 above)
Comment (tuntisz) [reply spec/services/provider_notifications.rb:26]: `deliver_later` enqueues a job with `deliver_now` as the method to call. So the test here ensures that there's a job queued up that calls `deliver_now`

**Review by tuntisz — APPROVED** [review]

File: spec/services/provider_notifications.rb (hunk 3 above)
Comment (jamestran201) [reply spec/services/provider_notifications.rb:26]: Oh I see, thanks!

**Pull request merged**